PACKAGE = "l3cam_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameters
# gen.add(name, paramtype, level, description, default, min, max, edit_method)

# CHANGE_POINTCLOUD_COLOR
change_pointcloud_color_enum = gen.enum([gen.const("RAINBOW",              int_t, 0, "RAINBOW"),
                                         gen.const("RAINBOW_Z",            int_t, 1, "RAINBOW_Z"),
                                         gen.const("INTENSITY",            int_t, 2, "INTENSITY"),
                                         gen.const("RGB_FUSION",           int_t, 3, "RGB_FUSION"),
                                         gen.const("POLARIMETRIC_FUSION",  int_t, 4, "POLARIMETRIC_FUSION"),
                                         gen.const("POL_PROCESSED_FUSION", int_t, 5, "POL_PROCESSED_FUSION"),
                                         gen.const("THERMAL_FUSION",       int_t, 6, "THERMAL_FUSION"),
                                         gen.const("RGBT_FUSION",          int_t, 7, "RGBT_FUSION")],
                                        "Enum to set pointcloud color")
gen.add("change_pointcloud_color", int_t, 0, "Change pointcloud color", 0, 0, 7, edit_method=change_pointcloud_color_enum)
# CHANGE_POINTCLOUD_COLOR_RANGE
gen.add("change_pointcloud_color_range_minimum", int_t, 1, "Change pointcloud color range minimum", 0, 0, 400000)
gen.add("change_pointcloud_color_range_maximum", int_t, 2, "Change pointcloud color range maximum", 400000, 0, 400000)
# CHANGE_DISTANCE_RANGE
gen.add("change_distance_range_minimum", int_t, 3, "Change distance range minimum", 0, 0, 400000)
gen.add("change_distance_range_maximum", int_t, 4, "Change distance range maximum", 400000, 0, 400000)

exit(gen.generate(PACKAGE, "l3cam_ros", "Pointcloud"))
